ecommerce/ecommerce/models.py

```diff
@@ -0,0 +1,48 @@
+from django.db import models
+
+class Image(models.Model):
+    image = models.ImageField(upload_to='products/')
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+    
+    def __str__(self):
+        return self.image.name
+
+class Product(models.Model):
+    name = models.CharField(max_length=255)
+    description = models.TextField()
+    price = models.DecimalField(max_digits=10, decimal_places=2)
+    stock = models.IntegerField()
+    images = models.ManyToManyField(Image, related_name='products')
```

Comment:
This would be one to many relationship. To achieve it you can put `product = models.ForeignKey(...)` in `Image` model.